### PR TITLE
feat: include task count in diary notifications

### PR DESCRIPTION
**Pull request by iProdigy**
Follow-up from https://github.com/pajlads/DinkPlugin/pull/321#issuecomment-1742133836

image

### CODE REVIEW

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java [hunk 1]
```diff
@@ -1,13 +1,68 @@
...
+     */
+    int areaTasksTotal;
+
+    @Override
+    public List<Field> getFields() {
+        List<Field> fields = new ArrayList<>(2);
+
+        if (areaTasksCompleted > 0 && areaTasksTotal > 0) {
+            fields.add(
+                new Field("Area Task Progress", Field.formatProgress(areaTasksCompleted, areaTasksTotal))
```
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:56]: Should `Area Task Progress` be rephrased to include the specific area? For example: `Karamja Progress`

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java [hunk 2]
```diff
@@ -1,13 +1,68 @@
...
+            );
+        }
+
+        if (tasksCompleted > 0 && tasksTotal > 0) {
+            fields.add(
+                new Field("Overall Task Progress", Field.formatProgress(tasksCompleted, tasksTotal))
```
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:62]: Should `Overall Task Progress` be rephrased?

File: src/main/java/dinkplugin/domain/AchievementDiary.java [hunk 3]
```diff
@@ -4,72 +4,72 @@
...
+    KOUREND_HARD(Varbits.DIARY_KOUREND_HARD, 11, "Kourend & Kebos", Difficulty.HARD),
+    KOUREND_ELITE(Varbits.DIARY_KOUREND_ELITE, 11, "Kourend & Kebos", Difficulty.ELITE),
+    LUMBRIDGE_EASY(Varbits.DIARY_LUMBRIDGE_EASY, 6, "Lumbridge & Draynor", Difficulty.EASY),
+    LUMBRIDGE_MEDIUM(Varbits.DIARY_LUMBRIDGE_MEDIUM, 6, "Lumbridge & Draynor", Difficulty.MEDIUM),
+    LUMBRIDGE_HARD(Varbits.DIARY_LUMBRIDGE_HARD, 6, "Lumbridge & Draynor", Difficulty.HARD),
+    LUMBRIDGE_ELITE(Varbits.DIARY_LUMBRIDGE_ELITE, 6, "Lumbridge & Draynor", Difficulty.ELITE),
+    MORYTANIA_EASY(Varbits.DIARY_MORYTANIA_EASY, 7, "Lumbridge & Draynor", Difficulty.EASY),
+    MORYTANIA_MEDIUM(Varbits.DIARY_MORYTANIA_MEDIUM, 7, "Lumbridge & Draynor", Difficulty.MEDIUM),
+    MORYTANIA_HARD(Varbits.DIARY_MORYTANIA_HARD, 7, "Lumbridge & Draynor", Difficulty.HARD),
+    MORYTANIA_ELITE(Varbits.DIARY_MORYTANIA_ELITE, 7, "Lumbridge & Draynor", Difficulty.ELITE),
```
Comment (pajlada) [inline src/main/java/dinkplugin/domain/AchievementDiary.java:52]: morytania

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java [hunk 4]
```diff
@@ -1,13 +1,68 @@
...
     int total;
+
+    /**
+     * The number of diary tasks that have been completed (across all regions and difficulties).
+     */
+    int tasksCompleted;
+
+    /**
+     * The total number of diary tasks within the game (across all regions and difficulties).
+     */
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:37]: remove the (s from the comments, imo. reads better without

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java (hunk 1 above)
Comment (pajlada) [reply src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:56]: I'm fine to leave this as is :+1:

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java (hunk 2 above)
Comment (pajlada) [reply src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:62]: I'm fine with this :+1:

**Review by pajlada — APPROVED** [review]
Looks good to me, haven't tested in game

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java (hunk 1 above)
Comment (Felanbird) [reply src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:56]: It's probablyyyyyyyyyyyyyy better to say the zone, `Karamja Progress` makes the most sense to me.

File: src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java (hunk 2 above)
Comment (Felanbird) [reply src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java:62]: overall is fine but we'll probably make it `Overall Progress` to go with the other suggestion

**Comment by pajlada** [comment]
finished Desert Medium Diary

image


```json
{
  "type": "ACHIEVEMENT_DIARY",
  "playerName": "pajdonk",
  "accountType": "NORMAL",
  "dinkAccountHash": "9748bae266b7923da080b8e5fc9f3b8855a7af49321bbf909b5a8ad9",
  "extra": {
    "area": "Desert",
    "difficulty": "MEDIUM",
    "total": 18,
    "tasksCompleted": 247,
    "tasksTotal": 492,
    "areaTasksCompleted": 25,
    "areaTasksTotal": 39
  },
  "discordUser": {
    "id": "85699361769553920",
    "name": "pajlada",
    "avatarHash": "e75df3dbe6cb04b3c9f0e090b3adb190"
  },
  "embeds": [
    {
      "title": "Achievement Diary",
      "description": "pajdonk has completed the Medium [Desert](https://oldschool.runescape.wiki/w/Special:Search?search=Desert%20Diary) Achievement Diary, for a total of 18 diaries completed",
      "author": {
        "name": "pajdonk",
        "url": "https://secure.runescape.com/m=hiscore_oldschool/hiscorepersonal?user1=pajdonk"
      },
      "color": 15990936,
      "image": {
        "url": "attachment://achievementDiaryImage.png"
      },
      "thumbnail": {
        "url": "https://oldschool.runescape.wiki/images/Achievement_Diaries_icon.png"
      },
      "fields": [
        {
          "name": "Desert Progress",
          "value": "```\n25/39 (64.1%)\n```",
          "inline": true
        },
        {
          "name": "Overall Progress",
          "value": "```\n247/492 (50.2%)\n```",
          "inline": true
        }
      ],
      "footer": {
        "text": "Powered by Dink",
        "icon_url": "https://github.com/pajlads/DinkPlugin/raw/master/icon.png"
      },
      "timestamp": "2023-10-25T21:46:14.913234269Z"
    }
  ]
}
```

**Pull request merged**